### PR TITLE
Create testing module with ContentMatcher. #54

### PR DESCRIPTION
**Pull request by lemonboston**

### CODE REVIEW

**Comment by dmfs** [comment]
please rename the module to `ContentTestPal`. We'll probably have another test module which provides tools for our internal testing, like certain matchers for `ContentProviderOperation.Builder`

**Comment by lemonboston** [comment]
Renamed module and package to `contenttestpal`.

File: contenttestpal/src/main/java/org/dmfs/android/contenttestpal/ContentMatcher.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+
+    public static Matcher<OperationsBatch> resultsIn(OperationsQueue queue, OperationsBatch assertingBatch)
+    {
+        return new ContentMatcher(queue, assertingBatch);
+    }
+
+
+    @SafeVarargs
+    public static Matcher<OperationsBatch> resultsIn(OperationsQueue queue, Operation... assertOperations)
```
Comment (dmfs) [inline contenttestpal/src/main/java/org/dmfs/android/contenttestpal/ContentMatcher.java]: `@SafeVarargs` should not be applied to non-generic parameters. Same below.

Comment (lemonboston) [reply contenttestpal/src/main/java/org/dmfs/android/contenttestpal/ContentMatcher.java]: Removed them.

**Comment by lemonboston** [comment]
Ready for review again.

**Review by dmfs — APPROVED** [review]

**Pull request merged**